enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,893 @@
+nodes is `True`) the cluster version operator will delete the `bundle-server`
+daemon set and verify that the metadata in all nodes (the content of the
+`metadata` field) is the same. If there are differences they will be reported
+as error conditions in the status of the `ClusterVersion` object and the
+upgrade process will be aborted. This is intended to prevent accidents like
+having two different USB sticks with different bundles plugged in two different
+nodes.
+
+Once the metadata has been validated the cluster version operator will ask the
+machine config operator to pin and pre-load the images specified in the
+metadata. To do so it will use the new `pinnedImages` and `imagesDirectory`
+fields of the `ContainerRuntimeConfig` object:
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: ContainerRuntimeConfig
+metadata:
+  name: pin-upgrade
+spec:
+  containerRuntimeConfig:
+    pinnedImages:
+    - quay.io/openshift-release-dev/ocp-release@sha256:...
+    - quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...
+    - quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...
+    imagesDirectory: /var/lib/upgrade/4.13.7-x86_64
+```
+
+The machine config operator will react to that pinning and pre-loading the
+images in all the nodes of the cluster, as described in the previous section.
+
+When the all the images are pinned and pre-loaded the cluster version operator
+will check the `spec.desiredUpdate.schedule.scheduleType` field of the
+`ClusterVersion` object. This field will be used to indicate if the upgrade has
+to be started manually later or it if should wait till the user explicitly sets
+it to `Immediate`. The default value for that will be `Immediate`. This is
+intended for situations where the user wishes to prepare everything for the
+upgrade in advance, but wants to perform the upgrade later.
+
+When the value of the `spec.desiredUpdate.schedule.scheduleType` field is
+`Immediate` the cluster version operator will trigger the regular upgrade
+process.
+
+When the upgrade has completed successfully the cluster version operator will
+will start a new `bundle-cleaner` job in each node that will clean all the
+artifacts potentially left around by other pieces of the upgrade.  In
+particular it will remove the `/var/lib/upgrade/4.13.7-x86_64` directory
+created by the `bundle-extractor`.
+
+#### Tool to create the upgrade bundle
+
+The upgrade bundle will be created by an engineer using a new `oc adm upgrade
+create bundle` command. This engineer will first determine the target version
+number, for example 4.13.7. Note that doing this will probably require access
+to the upgrade service available in api.openshift.com. Finding that upgrade
+version number is outside of the scope of this enhancement.
+
+The engineer will then need internet access and a Linux machine where she can
+run `oc adm upgrade create bundle`, for example:
+
+```bash
+$ oc adm upgrade create bundle \
+--arch=x86_64 \
+--version=4.13.7 \
```

Comment:
Limiting this to a version and arch as input is going to make it challenging to create bundles with hot fixes. It might be better to use an explicit image reference here, just like with other admin commands that operate on images.